tests/test_views.py

```diff
@@ -632,6 +632,79 @@ def test_detail_view_staff_all_users(self, mock_update):
         assert mock_update.call_count == 0
         assert models.StaleCompletion.objects.filter(resolved=False).count() == 2
 
+    @ddt.data(0, 1)
+    @XBlock.register_temp_plugin(StubCourse, 'course')
+    @XBlock.register_temp_plugin(StubSequential, 'sequential')
+    @XBlock.register_temp_plugin(StubHTML, 'html')
+    def test_detail_view_staff_requested_multiple_users(self, version):
+        """
+        Test that requesting course completions for a set of users filters out the other enrolled users
+        """
+
+        some_user = User.objects.create(username='test_user_2')
+        self.create_enrollment(
+            user=some_user,
+            course_id=self.course_key,
+        )
+        models.Aggregator.objects.submit_completion(
+            user=some_user,
+            course_key=self.course_key,
+            block_key=self.course_key.make_usage_key(block_type='course', block_id='course'),
+            aggregation_name='course',
+            earned=3.0,
+            possible=12.0,
+            last_modified=timezone.now(),
+        )
+
+        some_other_user = User.objects.create(username='test_user_3')
+        self.create_enrollment(
+            user=some_other_user,
+            course_id=self.course_key,
+        )
+        models.Aggregator.objects.submit_completion(
+            user=some_other_user,
+            course_key=self.course_key,
+            block_key=self.course_key.make_usage_key(block_type='course', block_id='course'),
+            aggregation_name='course',
+            earned=9.0,
+            possible=12.0,
+            last_modified=timezone.now(),
+        )
+
+        yet_another_user = User.objects.create(username='test_user_4')
+        self.create_enrollment(
+            user=yet_another_user,
+            course_id=self.course_key,
+        )
+        models.Aggregator.objects.submit_completion(
+            user=yet_another_user,
+            course_key=self.course_key,
+            block_key=self.course_key.make_usage_key(block_type='course', block_id='course'),
+            aggregation_name='course',
+            earned=6.0,
+            possible=12.0,
+            last_modified=timezone.now(),
+        )
+
+        self.client.force_authenticate(self.staff_user)
+        user_ids = str(some_user.id) + "," + str(yet_another_user.id)
+        response = self.client.get(self.get_detail_url(version, self.course_key, user_ids=user_ids))
+        self.assertEqual(response.status_code, 200)
```

Comment:
@Agrendalath I checked the logs and I'm getting an assertion error at this point, but I can't dig any deeper without the ability to run the test myself (which I can't do right now). Could you have a look and see if you find the error?